portality/models/article.py

```diff
@@ -17,20 +18,47 @@ def duplicates(cls, issns=None, publisher_record_id=None, doi=None, fulltexts=No
         issns = issns if isinstance(issns, list) else []
         urls = fulltexts if isinstance(fulltexts, list) else [fulltexts] if isinstance(fulltexts, str) or isinstance(fulltexts, unicode) else []
 
-        q = DuplicateArticleQuery(issns=issns,
-                                    publisher_record_id=publisher_record_id,
-                                    doi=doi,
-                                    urls=urls,
-                                    title=title,
-                                    volume=volume,
-                                    number=number,
-                                    start=start,
-                                    should_match=should_match)
-        # print json.dumps(q.query())
-
-        res = cls.query(q=q.query())
-        articles = [cls(**hit.get("_source")) for hit in res.get("hits", {}).get("hits", [])]
-        return articles
+        # in order to make sure we don't send too many terms to the ES query, break the issn list down into chunks
+        terms_limit = app.config.get("ES_TERMS_LIMIT", 1024)
+        issn_groups = []
+        lower = 0
+        upper = terms_limit
+        while lower < len(issns):
+            issn_groups.append(issns[lower:upper])
+            lower = upper
+            upper = lower + terms_limit
+
+        if issns is not None and len(issns) > 0:
```

Comment:
len(issns) > 0 should be enough, there's a type check and enforcement on line 18 above.